packages/salesforcedx-vscode-apex/src/commands/forceApexTestRunCodeAction.ts

```diff
@@ -23,21 +25,81 @@ const SfdxCommandlet = sfdxCoreExports.SfdxCommandlet;
 const SfdxWorkspaceChecker = sfdxCoreExports.SfdxWorkspaceChecker;
 const SfdxCommandletExecutor = sfdxCoreExports.SfdxCommandletExecutor;
 const notificationService = sfdxCoreExports.notificationService;
+const LibraryCommandletExecutor = sfdxCoreExports.LibraryCommandletExecutor;
+const channelService = sfdxCoreExports.channelService;
```

Comment:
Since the `LibraryCommandlet` executor currently lives in the core package, the channelService it uses is the default `Salesforce CLI` channel. If we switch to using the new `channelService` we have in the utils package right now, the experience is wonky because the `Starting command.../Ending command...` text appears in the default `Salesforce CLI` channel, while the output from the test run appears in the new channel that we've created. And since the command palette portion of the `test:run` functionality still lives in the core package as well, all the output from that command is displayed via the `Salesforce CLI` channel too

In a follow up PR, I think we should migrate the LibraryCommandlet to the utils package, the Apex commands (test:run command) to the apex extension, and then switch this command to use the new channelService so it doesn't cause customers any confusion on which channel they should be looking at for apex test results

Reply:
Yes, that is part of the work we'll be doing in January to address some perf issues in the Apex extension.